test/tank_test.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tank_manager/model/tank.dart';
+import 'dart:convert';
+
+void main() {
+  test('Tank set name/ip', () {
+    final tank = Tank('tank_test', '192.168.0.1');
+
+    expect(tank.name, 'tank_test');
+    expect(tank.ip, '192.168.0.1');
+  });
+
+  test('Tank equivalence', () {
+    final tank1 = Tank('tank_test', '192.168.0.1');
+    final tank2 = Tank('tank_test', '192.168.0.1');
+    final tank3 = Tank('tank', '192.168.0.1');
+
+    expect(tank1, tank2);
+    expect(tank1, isNot(tank3));
```

Comment:
isNot() uses the overloaded operator == to check equivalence. This means tank1 is not tank2 if the name attribute differs. For instance, `{ expect(tank1, isNot(tank1)); }`  will fail because both tanks are named 'tank_test'. Also, `{ expect(tank1, isNot(tank2)); }` will fail because both have the same name.